workshops/chat/README.md

```diff
@@ -0,0 +1,258 @@
+
+```js
+function pushMessage(event) {
+  if (event.keyCode == 13) {
+  }
+}
+```
+
+We need the message's sender and text, so that we can pass them to Firebase. We'll extract the values from the HTML inputs, and store them in variables:
+
+```js
+function pushMessage(event) {
+  if (event.keyCode == 13) {
+    var name = $('#nameInput').val();
+    var text = $('#messageInput').val();
+  }
+}
+```
+
+Finally we'll push to Firebase by using `.push()` method on the reference we've created:
+
+```js
+function pushMessage(event) {
+  if (event.keyCode == 13) {
+    var name = $('#nameInput').val();
+    var text = $('#messageInput').val();
+    myDataRef.push({name: name, text: text});
+  }
+}
+```
+
+And to finish it up, we'll clear the message input:
+
+```js
+function pushMessage(event) {
+  if (event.keyCode == 13) {
+    var name = $('#nameInput').val();
+    var text = $('#messageInput').val();
+    myDataRef.push({name: name, text: text});
+    $('#messageInput').val('');
+  }
+}
+```
+
+### Retrieving Messages from Firebase
+
+We want the message we send to be displayed immediately after we send it, so we'll use a the `.on()` method of our reference to say that "upon adding a message to Firebase, do something." The thing that we'd like to do is call a function to show the message. We'll also create this function now:
+
+```js
+myDataRef.on("child_added", showMessage);
+
+function showMessage(snapshot) {
+
+}
+```
+
+Note that we are passing an argument called `snapshot` into `showMessage`. This is how Firebase returns the message's data. We can access said data by calling one of its methods (`.val()`). We'll store this data in a variable named `message`:
+
+```js
+function showMessage(snapshot) {
+  var message = snapshot.val();
+}
+```
+
+Now that we've got the message data in `message`, we'll extract the specific pieces of message data (sender name and message text), and store them in their own variables:
+
+```js
+function showMessage(snapshot) {
+  var message = snapshot.val();
+  var messageSender = message.name;
+  var messageContent = message.text;
+}
+```
+
+Lastly, we'll add it to our HTML using some jQuery sweetness. Start by creating a new element to store the entire message. We can also add a class to this element to classify it as a message.
+
+```js
+function showMessage(snapshot) {
+  var message = snapshot.val();
+  var messageSender = message.name;
+  var messageContent = message.text;
+  var messageDiv = $("<div/>").addClass("message");
+}
+```
+
+Then we'll create two [`span`](TODO) elements to store the sender and message content, and fill them with the appropriate info.
+
+```js
+function showMessage(snapshot) {
+  var message = snapshot.val();
+  var messageSender = message.name;
+  var messageContent = message.text;
+
+  var messageDiv = $("<div/>");
+  var senderDiv = $("<span/>").text(messageSender + ": "));
```

Comment:
I think you have an extra parenthesis on this line.

What does `<span/>` mean in the jQuery function? What's actually happening here? Why is it called `senderDiv` instead of `senderSpan`?